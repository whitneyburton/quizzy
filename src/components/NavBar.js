import React, { Component } from 'react';
import '../styles/main.scss';

export default class NavBar extends Component {
  constructor() {
    super();
    this.state = {

    }
  }
  render() {
    let { setCategory } = this.props;
    return (
      <div className="NavBar">
        <h1 className="quizzy-title">QUIZZY!</h1>
        <div className="filter-category-buttons">
          <p className="choose-category">Choose an array prototype category:</p>
          <button
            onClick={setCategory}
            className="mutator-button buttons"
            type="button">Mutator</button>
          <button
            onClick={setCategory}            
            className="accessor-button buttons"
            type="button">Accessor</button>
          <button
            onClick={setCategory}
            className="iteration-button buttons"
            type="button">Iteration</button>
          <button
            onClick={setCategory}
            className="all-methods-button buttons"
            type="button">All Methods!</button>
        </div>

      </div>
    );
  }
}