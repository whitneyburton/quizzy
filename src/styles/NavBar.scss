.NavBar {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: linear-gradient(#0c4640, $deep-teal);
  padding: 20px 0px 30px 0px;
  
  .quizzy-title {
    font-size: 4rem;
    margin: 0px;
    padding-bottom: 20px;
    color: #fdf79b;
    letter-spacing: 15px;
  }
  
  .filter-section {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 70%;

    .choose-category {
      font-weight: bold;
      color: #FFF;
      font-size: 16px;
    }
    
    .filter-button {
      padding: 5px;
      margin: 0px 10px;
      width: 25%;
    }
  }
}

.buttons {
  font-size: 13px;
  border-radius: 10px;
  transition: all 300ms ease;
  
  &:hover {
    cursor: pointer;
    background-color: $whitesmoke;
    box-shadow: 1px 1.5px 1px #9b9b9b;
  }
  &:focus {
    outline: 0;
  }
}

@media screen and (max-width: 755px) {
  .NavBar {
    padding: 10px 0px;
    
    .filter-section {
      flex-direction: column;

      .filter-button {
        margin: 10px 0px;
        width: 70%;
      }
    
      .choose-category {
        text-align: center;
      }
    }
  }

  .buttons {
    font-size: 12px;
  }
}