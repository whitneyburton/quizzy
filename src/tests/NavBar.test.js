import React from 'react';
import NavBar from '../Components/NavBar';
import { shallow } from 'enzyme';

describe('NavBar', () => {
  let wrapper;
  const filterCardsByCategoryMock = jest.fn();
  const updateCategoryMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <NavBar
      filterCardsByCategory={filterCardsByCategoryMock}
      updateCategory={updateCategoryMock}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})