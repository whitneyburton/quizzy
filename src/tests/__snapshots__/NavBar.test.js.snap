// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<div
  className="NavBar"
>
  <h1
    className="quizzy-title"
  >
    QUIZZY!
  </h1>
  <div
    className="filter-section"
  >
    <p
      className="choose-category"
    >
      Choose an array prototype category:
    </p>
    <button
      className="buttons filter-button"
      onClick={[Function]}
      type="button"
    >
      Mutator
    </button>
    <button
      className="buttons filter-button"
      onClick={[Function]}
      type="button"
    >
      Accessor
    </button>
    <button
      className="buttons filter-button"
      onClick={[Function]}
      type="button"
    >
      Iteration
    </button>
    <button
      className="buttons filter-button"
      onClick={[Function]}
      type="button"
    >
      All!
    </button>
  </div>
</div>
`;
